### PR TITLE
DEVX-2527: Migrate client image from local build to stored in docker hub

### PR DESCRIPTION
**Pull request by awalther28**
### Description 

 https://confluentinc.atlassian.net/browse/DEVX-2527

_What behavior does this PR change, and why?_
Update docs, docker-compose, and script to use client image that is pushed up to docker hub. Add makefile to make pushing and creating images simple. Update readme to talk about readme. 

```
localhost:ccloud-monitoring awalther$ docker system prune -a
... <deletes all images>
Total reclaimed space: 26.81GB

localhost:ccloud-monitoring awalther$ time docker-compose up -d 
... <pulls all images> ...
real	0m29.466s
user	0m0.840s
sys	0m0.222s
```
Granted, I have pretty speedy internet speeds 😁 

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

- [x] Documentation 
- [x] ccloud-monitoring

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

- [ ] Documentation 
- [ ] ccloud-monitoring

### CODE REVIEW

File: ccloud-monitoring/README.md [hunk 1]
```diff
@@ -5,3 +5,6 @@
 # Documentation
 
 You can find the documentation and instructions for running this Monitoring Apache Kafka Clients to Confluent Cloud example at [https://docs.confluent.io/platform/current/tutorials/examples/ccloud-monitoring/docs/index.html](https://docs.confluent.io/platform/current/tutorials/examples/ccloud-monitoring/docs/index.html?utm_source=github&utm_medium=example&utm_campaign=tm.devx_ch.examples_content.ccloud-monitoring#ccloud-monitoring-index).
+
+## Client Image
+The client docker image is stored in docker hub. If dependencies change for the client (ie the pom.xml changes), the client docker image will need to be recreated locally and pushed to docker hub. You can do this by running `make push-image`.
```
Comment (rspurgeon) [inline ccloud-monitoring/README.md]: Public users won't have access to push to cnfldemos, so you might want to clarify who will be allowed to run this command.  If this is meant for public users to be able to do, then we probably need to clarify in the docs somewhere how to build and override the local usage of the image.

Comment (awalther28) [reply ccloud-monitoring/README.md]: Great point. I removed that note and put it our[ internal wiki page about "releases" ](https://confluentinc.atlassian.net/wiki/spaces/DEVX/pages/1809390195/How+To+Prep+Our+Repos+For+a+CP+Release#examples).

**Review by ybyzek — APPROVED** [review]
@awalther28 Two comments for consideration:

1. Rename image from `*monitoring*` to `*observability*`?
2. Maybe a note _somewhere_ that says the src code in the project is built into that Docker image?

I defer to you on if and how to make those changes, and approving LGTM in the meanwhile.

**Comment by awalther28** [comment]
Thank you @ybyzek . I updated the docker image name, pushed up a new image, and validated that everything comes up as expected. Also I added a note in the docker-compose about the src code and added a note in our internal docs. 
Next step is to rename "everything" to observability.

**Pull request merged**